src/routes/safe/components/Transactions/TxList/modals/ApproveTxModal.tsx

```diff
@@ -200,6 +200,8 @@ export const ApproveTxModal = ({ onClose, isOpen, transaction }: Props): React.R
   const safeAddress = extractSafeAddress()
   const txInfo = useTxInfo(transaction)
 
+  const submitDisabled = sameAddress(userAddress, safeAddress)
```

Comment:
I would suggest we use the `sameAddressAsSafeSelector` selector here instead so that the logic is centralised.